spclient/spclient.go

```diff
@@ -145,34 +146,43 @@ func (c *Spclient) PutConnectState(spotConnId string, reqProto *connectpb.PutSta
 	if err != nil {
 		return fmt.Errorf("failed marshalling PutStateRequest: %w", err)
 	}
-
-	resp, err := c.request(
-		"PUT",
-		fmt.Sprintf("/connect-state/v1/devices/%s", c.deviceId),
-		nil,
-		http.Header{
-			"X-Spotify-Connection-Id": []string{spotConnId},
-			"Content-Type":            []string{"application/x-protobuf"},
-		},
-		reqBody,
-	)
+	_, err = backoff.RetryWithData(func() (*http.Response, error) {
+		resp, err := c.request(
+			"PUT",
+			fmt.Sprintf("/connect-state/v1/devices/%s", c.deviceId),
+			nil,
+			http.Header{
+				"X-Spotify-Connection-Id": []string{spotConnId},
+				"Content-Type":            []string{"application/x-protobuf"},
+			},
+			reqBody,
+		)
+		if err != nil {
+			return nil, err
+		}
+		defer func() { _ = resp.Body.Close() }()
+
+		if resp.StatusCode != 200 {
+			var putError putStateError
+			if resp.ContentLength <= 0 {
+				log.Debugf("failure to decode response.Body, ContentLength was: %d", resp.ContentLength)
+				return nil, fmt.Errorf("failure to decode response.Body, ContentLength was: %d", resp.ContentLength)
+			}
```

Comment:
Isn't this redundant as the EOF will be caught below when decoding the JSON?

Reply:
You are right, it is indeed reduntant. I have removed that part of code